packages/puppeteer-core/src/common/Coverage.ts

```diff
@@ -74,6 +74,11 @@ export interface JSCoverageOptions {
    * Whether the result includes raw V8 script coverage entries.
    */
   includeRawScriptCoverage?: boolean;
+  /**
+   * Whether to collect coverage information at the function level.
+   * If false, coverage will be collected at the block level.
+   */
+  usePerFunctionScope?: boolean;
```

Comment:
Actually, could you name this `useBlockCoverage` with a default of `true`. Not sure if people will understand `usePerFunctionScope`. They will with `useBlockCoverage` because that's what the docs call it (https://chromedevtools.github.io/devtools-protocol/tot/Profiler/)